modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/keystore/management/v1/KeystoreManagementSuccessTest.java

```diff
@@ -0,0 +1,192 @@
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.identity.integration.test.rest.api.server.keystore.management.v1.model.CertificateResponse;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Tests for successful cases of the Keystore Management REST API.
+ */
+public class KeystoreManagementSuccessTest extends KeystoreManagementBaseTest {
+
+    private static final String SUPER_TENANT_PUBLIC_CERT_ALIAS = "wso2carbon";
+
+    @Factory(dataProvider = "restAPIUserConfigProvider")
+    public KeystoreManagementSuccessTest(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        this.context = isServer;
+        this.authenticatingUserName = context.getContextTenant().getTenantAdmin().getUserName();
+        this.authenticatingCredential = context.getContextTenant().getTenantAdmin().getPassword();
+        this.tenant = context.getContextTenant().getDomain();
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws IOException {
+
+        super.testInit(API_VERSION, swaggerDefinition, tenant);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Test
+    public void testGetAllCertificateAliases() throws IOException, URISyntaxException {
+
+        Response response = getResponseOfGet(KEYSTORE_MANAGEMENT_API_BASE_PATH +
+                KEYSTORE_MANAGEMENT_API_CERTIFICATE_PATH);
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+        ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+        List<CertificateResponse> responseFound =
+                Arrays.asList(jsonWriter.readValue(response.asString(), CertificateResponse[].class));
+        Assert.assertNotNull(responseFound);
+        if (StringUtils.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, tenant)) {
+            Assert.assertTrue(responseFound.stream().anyMatch(certificateResponse -> certificateResponse.getAlias()
```

Comment:
Rather than using de-serialization to object, you can use JSON paths. Refer https://github.com/rest-assured/rest-assured/wiki/Usage